bin/ch-run.c

```diff
@@ -646,3 +699,127 @@ char *storage_default(void)
 
    return storage;
 }
+
+void readJSONFile(const char *filename, struct args *args) {
+
+    FILE *file = fopen(filename, "rb");
+    char *buffer = NULL;
+    size_t length;
+    size_t file_len;
+    
+    if (file == NULL) {
+        fprintf(stderr, "Error: could not open file %s\n", filename);
+        return;
+    }
+
+    // Get the file length
+    fseek(file, 0, SEEK_END);
+    length = ftell(file);
+    fseek(file, 0, SEEK_SET);
+    // Allocate content buffer
+    buffer = malloc(length);
+    file_len = fread(buffer, 1, length, file);
+    printf("Read %lu bytes of %lu\n", file_len, length);
+
+    fclose(file);
+
+    // Parse the JSON data
+    cJSON *json = cJSON_Parse(buffer);
+    // Check if parsing was successful
+    if (json == NULL) {
+        const char *error_ptr = cJSON_GetErrorPtr();
+        if (error_ptr != NULL) {
+            fprintf(stderr, "Error before: %s\n", error_ptr);
+        }
+        goto end;
+    }
+
+    // Print the JSON data
+    char *json_tree = cJSON_Print(json);
+    if (json_tree == NULL)
+    {
+        fprintf(stderr, "Failed to print JSON tree.\n");
+    }
+    //printf("%s\n", json_tree);
+
+    // Process the JSON data
+    visit(oci_top, json, args);
+
+    // Clean up
+end:
+    cJSON_Delete(json);
+    free(buffer);
+}
+
+void dispatch(struct json_dispatch action, cJSON *json, struct args *args) {
+    if (action.f != NULL)
+        action.f(json, args);
+    if (action.children != NULL)
+        visit(action.children, json, args);
+}
+
+
+void visit(struct json_dispatch actions[], cJSON *json, struct args *args) {
+    for (int i =0; actions[i].name != NULL; i++) {
+        cJSON *subtree = cJSON_GetObjectItem(json, actions[i].name);
+        if (cJSON_IsArray(subtree)) {
+            cJSON *elem;
+            cJSON_ArrayForEach(elem, subtree)
+                dispatch(actions[i], elem, args);
+        } else {
+            dispatch(actions[i], subtree, args);
+        }
+    }
+}
+
+void oci_ociVersion(cJSON *json, struct args *args) {
+   // Need to validate this version. Not sure how just yet.
+   printf("oci_ociVersion: %s\n", json->valuestring);
+}
+
+
+void oci_env(cJSON *json, struct args *args) {
+   // TO DO: corresponds to --set-env
+   printf("oci_env: %s\n", json->valuestring);
+}
+
+
+void oci_cwd(cJSON *json, struct args *args) {
+   // corresponds to --cd
+   //printf("oci_cwd: %s\n", json->valuestring);
+   if (args->initial_dir == NULL) {
+      args->initial_dir = json->valuestring;
+   }
+   //printf("after assignment in oci_cwd: %c\n",json->valuestring);
+   printf("oci_cwd is: %s\n", args->initial_dir);
+}
+
+
+void oci_path(cJSON *json, struct args *args) {
+   //printf("oci_path: %s\n", json->valuestring);
+
+   args->c.newroot = json->valuestring;
+
+   if (!path_exists(args->c.newroot, NULL, true)) {
+      args->c.newroot = path_join(args->c.img_ref, json->valuestring);
```

Comment:
This assignment works, but I believe it's because I'm using the function ```path_join()``` and it's storing the pointer values the right way.